hnn_core/cell.py

```diff
@@ -25,6 +25,68 @@ def _get_cos_theta(p_secs, sec_name_apical):
     return cos_thetas
 
 
+def _calculate_gaussian(x_val, height, lamtha):
```

Comment:
Along with one of @jasmainak's prior comments, I'm wondering if this function actually belongs in `network.py`. It's a relatively minor issue, but my vote is to make `cell.py` and `network.py` as siloed as possible s.t. the `cell` module contains functions and classes that deal solely with internal properties like morphology, synapses, etc.

Reply:
I like this is concept, but I think just moving this function now will make things less intuitive. `network_builder.py` calls which does the weight calculation. Moving this function would require an importing from `Network.py`.

I think a better path might be precomputing these weights before the network is built, or a least moving these functions to `network_builder.py` and geting rid of `Cell.parconnect_from_src` to your goal of separation. 

Perhaps save this for a MAINT PR?

Reply:
I thought of the precomputing but the issue is it will increase the size of our pickled object? Agree, this is too much for this PR. Btw, I don't think you need `_calculate_gaussian` as a separate function. It's used just once and is less than 3 lines. Nesting should be avoided when possible. Zen of Python:

```sh
$ python -c 'import this' | sed -n 5p
Flat is better than nested.
```

Reply:
Totally agree with the nesting, but I have plans to use that function immediately after this PR. Specifically for calculating a distance dependent connection probability.